code/modules/projectiles/ammunition.dm

```diff
@@ -97,8 +97,8 @@ They're all essentially identical when it comes to getting the job done.
 	if(!source.current_rounds)
 		to_chat(user, "<span class='warning'>\The [source] is empty.</span>")
 		return
-
-	if(!istype(source, /obj/item/ammo_magazine/handful)) //if it isn't a handful, give a delay.
+//using handfuls; and filling internal mags has no delay.
+	if(!istype(source, /obj/item/ammo_magazine/handful) && !istype(src, /obj/item/ammo_magazine/internal) ) 
```

Comment:
is this hacky?

Reply:
Have you tested this?
istype is a specific check it wont work for child objects.

You could instead try `ispath(type, /some/type/path`

Reply:
It seemingly works, haven't heard any complaints, but I'll swap it around anyway

Reply:
It may be you never do checks against other types and this is good enough.